src/app/models/buildings/building.ts

```diff
@@ -45,22 +47,47 @@ export class Building {
     return this.level + 1 <= this.maxLevel
   }
 
+  getLevelContainer() {
+    const levelContainer = document.querySelector(
+      `.building__level--${this.getTitle()}`
+    ) as HTMLElement
+    return levelContainer
+  }
+
+  getBuildingPriceContainer() {
+    const container = document.querySelectorAll(
+      `.building__price--${this.getTitle()}`
+    )
+    return container
+  }
+
+  updateBuildingContainer() {
+    let i = 0
+    this.getLevelContainer().textContent = `Level: ${this.level}`
+    this.getBuildingPriceContainer().forEach((element) => {
+      element.textContent = `${this.resourcesNeededToBuild[i].count} ${this.resourcesNeededToBuild[i].type}`
+      i++
+    })
+  }
+
   update() {
     if (this.isBuilding) {
       console.log(`${this} is building`)
-      this.timeToBuildInMiliseconds -= GAME_LOOP_DELAY_IN_MILISECONDS
-
-      if (this.timeToBuildInMiliseconds <= 0) {
+      this.remainingTimeToBuild -= GAME_LOOP_DELAY_IN_MILISECONDS
+      console.log(this.remainingTimeToBuild)
+      if (this.remainingTimeToBuild <= 0) {
         this.isBuilding = false
-        this.level++
-
         this.game.handleBuildingWasBuilt(this)
         this.handleBuildingWasBuilt()
       }
     }
   }
 
-  protected handleBuildingWasBuilt() {}
```

Comment:
create new protected method handlePassiveIncomeBuildingWasBuilt(), leave it empty and call it in update method. Change method names in sawmill quarry and goldmine to 'handlePassiveIncomeBuildingWasBuilt'